File: ios-catalogue/App/Playground/PreviewSurface.swift
import SwiftUI
import CoreUI

/// Container that surrounds the live-preview component.
/// Themed surface, subtle border, generous padding so taller components
/// can spread without the surface flicking between heights.
struct PreviewSurface<Content: View>: View {
    @Environment(\.colorScheme) private var scheme
    private let content: () -> Content

    init(@ViewBuilder content: @escaping () -> Content) {
        self.content = content
    }

    var body: some View {
        content()
            .frame(maxWidth: .infinity, minHeight: 200)
            .padding(PrismaSpacing.sp7)
            .background(PrismaSemanticColors.surfaceRaised.themed(scheme))
            .clipShape(RoundedRectangle(cornerRadius: PrismaRadius.lg))
            .overlay(
                RoundedRectangle(cornerRadius: PrismaRadius.lg)
                    .strokeBorder(PrismaSemanticColors.borderSubtle.themed(scheme), lineWidth: 1)
            )
    }
}

File: android-catalogue/catalogue/src/main/java/xyz/ksharma/prisma/catalogue/playground/PreviewSurface.kt
package xyz.ksharma.prisma.catalogue.playground

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.heightIn
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.unit.dp
import xyz.ksharma.prisma.coreui.themed
import xyz.ksharma.prisma.tokens.PrismaRadius
import xyz.ksharma.prisma.tokens.PrismaSemanticColors
import xyz.ksharma.prisma.tokens.PrismaSpacing

/**
 * Container the live-preview component renders inside.
 *
 * Centered, themed surface with a subtle border. Sized so taller components
 * (modals, sheets) can spread without the surface flicking between heights.
 */
@Composable
internal fun PreviewSurface(
    modifier: Modifier = Modifier,
    content: @Composable () -> Unit,
) {
    Box(
        modifier = modifier
            .fillMaxWidth()
            .heightIn(min = 200.dp)
            .clip(RoundedCornerShape(PrismaRadius.Lg))
            .background(PrismaSemanticColors.SurfaceRaised.themed())
            .border(
                1.dp,
                PrismaSemanticColors.BorderSubtle.themed(),
                RoundedCornerShape(PrismaRadius.Lg),
            )
            .padding(PrismaSpacing.Sp7),
        contentAlignment = Alignment.Center,
    ) {
        content()
    }
}
